docs/DeveloperGuide.md

```diff
@@ -1,33 +1,98 @@
 # Developer Guide
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+Ideas for this project partly comes from NUSmods at https://nusmods.com/
+
+API and library used includes the following:
+- [NUSMods API](https://api.nusmods.com/v2/) - To extract information about modules in NUS
+- [FasterXML](https://github.com/FasterXML/jackson) - Helps parse json files from the NUSMods API
 
+## Design & implementation
```

Comment:
Should add more Class diagram and Sequence diagram to this part to show developers how your app's components integrate